### DISCUSSION TITLE
Mixed content when deployed behind LB

### DISCUSSION THREAD

**Discussion by aimestereo** [coollabsio/coolify#4040] [Q&A] [answered]
### Description

I have a problem with Coolify admin. It is deployed on AWS EC2 and SSL termination happens before traffic reaches instance.

In my setup Coolify thinks that it's served via `http` and hardcode `http` into generated pages.

How do I teach Coolify that it's behind https? what am I missing?

Notes:

* To do any configurations I access site via `http://` by direct link. via `https://my.domain` i can' use it, since UI is completely broken

after I've created this issue, I've found [dns configuration](https://coolify.io/docs/knowledge-base/dns-configuration), so I've set `Settings -> General -> Instance's Domain` to my `https://my.domain` that I use to access instance - nothing changed, except this:

```html
Validating DNS failed.
<br><br>
Make sure you have added the DNS records correctly.
<br><br>
https://coolify.truagents.dev->host.docker.internal
<br><br>
Check this <a target='_blank' class='underline dark:text-white' href='https://coolify.io/docs/knowledge-base/dns-configuration'>documentation</a> for further help.
```

I've also tried to disabled Domain validation - it let me save settings, but it didn't help with overall problem

### Minimal Reproduction (if possible, example repository)

I have such deployment  in AWS:
`- https ->` CloudFront (terminate ssl) `- http ->` LoadBalancer -> EC2 with Coolify

### Exception or Error

Example: part of Server page - see a lot of `http://`:

```html
<title>Coolify</title>
<link rel="icon" href="http://coolify.truagents.dev/coolify-transparent.png" type="image/x-icon" />
...
<link rel="preload" as="style" href="http://coolify.truagents.dev/build/assets/app-f77813ca.css" />
<link rel="modulepreload" href="http://coolify.truagents.dev/build/assets/app-4ed993c7.js" />
<link rel="stylesheet" href="http://coolify.truagents.dev/build/assets/app-f77813ca.css" data-navigate-track="reload" />
<script type="module" src="http://coolify.truagents.dev/build/assets/app-4ed993c7.js" data-navigate-track="reload"></script> 
```

<img width="1085" alt="image" src="https://github.com/coollabsio/coolify/assets/2869535/e35fa7f1-267f-4cfc-8a55-494a07b45b63">

<img width="1082" alt="Screenshot 2024-06-13 at 08 55 19" src="https://github.com/coollabsio/coolify/assets/2869535/e35eeb3e-96ac-45ae-90e7-10596b02efe6">

### Version

v4.0.0-beta.294

**Comment by stuart-haas** [comment]
@aimestereo Have you considered using a network load balancer instead and terminating SSL on the server? This seems like a simple configuration issue between cloudfront and your ec2 instance. The server you are running should expose another port for 443 or 9443 and then you can forward that traffic to a network load balancer and/or use cloudfront in front of that to handle ssl traffic.

**Comment by aimestereo** [comment]
@stuart-haas, thanks, I will try it out, originally I just thought that I misconfigured Coolify, but it seems, for https to work correctly I need to let Coolify to terminate SSL

**Comment by bh0mbalziyad** [comment, marked as answer]
Can confirm that this is indeed solved by letting traefik terminate the SSL connection. I initially had Coolify deployed behind Nginx and wasn't using Traefik but that setup wasn't ideal as I was having to spend an unreasonable amount of time on rewriting my Nginx rules.

**Comment by aimestereo** [comment]
@bh0mbalziyad if you don't mind, can you share your Traefik config? I don't see much difference in how I terminate SSL: on ALB, Nginx or Traefik. 
The only difference for me was: Coolify terminates SSL vs other tool before traffic comes to Coolify

**Comment by bh0mbalziyad** [comment]
@aimestereo I'm using the default configuration which coolify autogenerates, the only change I've made was to add a dynamic configuration to include SSL certificates

```yaml
tls:
    certificates:
        - certFile: /traefik/certs/wildcard.cert
          keyFile: /traefik/certs/wildcard.key
```

**Comment by wp-coin** [comment]
this is definitely a bug in coolify config, it happens on new WordPress installs with default configurations

One approach is to use middleware to set the X-Forwarded-Proto header to https, which informs backend services that the request is secure.
- "traefik.http.middlewares.sslheader.headers.customrequestheaders.X-Forwarded-Proto=https"

Content Security Policy (CSP): You can also use a Content Security Policy to enforce HTTPS for all resources. This can be done by adding the following header in your Traefik configuration:
- "traefik.http.middlewares.csp.headers.contentsecuritypolicy=upgrade-insecure-requests"

**Comment by supun-io** [comment]
Same issue here with Cloudflare Proxy. If I set the domain in settings to https it redirects infinitely because it does not detect HTTPS from proxy headers. For now, I have disabled Cloudflare Proxy. This issue might need to be handled in Coolify.

**Reply by djsisson** [reply]
With cf proxy you need to make sure its set to strict, or at least not http in cf

Otherwise cf terninates ssl and only sends http request to your server. Which then redirects back to https causing the infinite loop.
